test/test_tensor_creation_ops.py

```diff
@@ -832,6 +832,111 @@ def compare_helper_(like_fn, t):
                     tp = t.permute(p)
                     compare_helper_(like_fn, tp)
 
+    @onlyOnCPUAndCUDA
+    @dtypes(*torch.testing.get_all_dtypes(include_bfloat16=False, include_bool=False))
+    def test_hsplit(self, device, dtype):
+        inputs = (
+            ((), 3),
+            ((), [2, 4, 6]),
+            ((6), 2),
+            ((6), 4),
+            ((6), [2, 5]),
+            ((6), [7, 9]),
+            ((3, 8), 4),
+            ((3, 8), 5),
+            ((3, 8), [1, 5]),
+            ((3, 8), [3, 8]),
+            ((S, S, S), 2),
+            ((S, S, S), [1, 4]),
+            ((S, 0, S), 3),
+            ((S, S, 0), [2, 6]),
+        )
+
+        for shape, arg in inputs:
+            t = make_tensor(shape, dtype, device)
+            t_n = t.cpu().numpy()
+
+            if isinstance(arg, list) or len(shape) == 0:
+                result = torch.hsplit(t, arg)
+                result_n = np.hsplit(t_n, arg)
+                self.assertEqual(result, result_n)
+            else:
+                dim = 1 if len(shape) > 1 else 0
+                if shape[dim] % arg == 0:
+                    result = torch.hsplit(t, arg)
+                    result_n = np.hsplit(t_n, arg)
+                    self.assertEqual(result, result_n)
+                else:
+                    with self.assertRaisesRegex(RuntimeError, "tensor split does not result in an equal division"):
```

Comment:
make sure regex here matches C++ message(with respect to regular expression rules.)